packages/vue-app/template/utils.js

```diff
@@ -597,6 +597,10 @@ function formatUrl (url, query) {
   let result = (protocol ? protocol + '://' : '//') + parts.shift()
 
   let path = parts.join('/')
+  if (path === '') {
+    return result + (parts.length === 1 ? '/' : '')
```

Comment:
This early return is not using incorporating the `query` parameter. I think your new logic here should be included in the logic on line 610 below (`result += path ? '/' + path : ''`).

Reply:
Here is only for url only has a domain without path, querystring and hash

Reply:
What about if you call redirect like this?

```javascript
<script>
export default {
  middleware ({ redirect }) {
    const query = { foo: 'bar' }
    return redirect('https://nuxtjs.org/', query)
  }
}
</script>
```

I mean, if you pass a second parameter to `redirect()`, like above, which is the `query` parameter, which is being ignored now. Please correct me if I'm wrong.

Reply:
Ah, correct, thanks